pkg/auth/dependency/webapp/state_middleware.go

```diff
@@ -17,6 +17,23 @@ func (m *StateMiddleware) Handle(next http.Handler) http.Handler {
 
 		q := r.URL.Query()
 		sid := q.Get("x_sid")
+
+		if sid == "" && r.Method == "GET" {
```

Comment:
Are we going to always generate a SID on entry GET, not waiting for user clicking next button?

Reply:
If we do it lazily, we need to bring back MayRestoreState.

GET: may restore state
POST entry step: always regenerate state
POST subsequent steps: must restore state

Doing it lazily is better IMO. Let's change it back?

Reply:
> Doing it lazily is better IMO. Let's change it back?

I agree. However, instead of having two function `MayRestoreStore` and `MustRestoreState`, can we have single `RestoreState` function and ignore not found error explicitly when needed?